src/main/kotlin/com/chattriggers/ctjs/engine/loader/ModuleUpdater.kt

```diff
@@ -97,26 +97,28 @@ object ModuleUpdater {
                 if (rootFolder.hasNext()) throw Exception("Too big")
 
                 val realName = moduleFolder.fileName.toString().trimEnd(File.separatorChar)
-                val targetPath = File(modulesFolder, realName).apply { mkdir() }.toPath()
+                File(modulesFolder, realName).apply { mkdir() }
 
-                Files.walk(moduleFolder).use { paths ->
-                    paths.forEach { path ->
-                        val relativePath = path.drop(1).joinToString(File.separatorChar.toString())
+                Files.walk(moduleFolder).forEach {
+                    path -> val resolvedPath = Paths.get(modulesFolder.toString(), path.toString())
 
-                        if (relativePath.isNotEmpty()) {
-                            Files.copy(path, targetPath.resolve(relativePath))
+                    if (Files.isDirectory(resolvedPath)) {
+                        if (!Files.exists(resolvedPath)) {
```

Comment:
You're walking the file tree, so this file has to exist, right?

Reply:
I'm walking the file tree of the newly downloaded module folder contained in the zip so the file referred to by resolvedPath does not necessarily exist in the case of importing modules. However, I also missed the fact that the directory function returns false if the file is either not a directory OR doesn't exist meaning the 2nd if statement ends up being redundant and never runs. Funnily enough this all works out with the copy function as well because if the passed in file to copy is a directory then it will automatically create the directory as an empty directory of the same name. So in the end everything worked out, although not as I thought it would. TLDR I'm getting rid of the 2nd if statement.